### PR TITLE
Allow backtick markdown in lesson plan descriptions

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Support markdown in progress lesson descriptions, but only after passing that markdown through a very restrictive renderer (i.e. it strips everything other than backticks).

image

### CODE REVIEW

File: dashboard/app/helpers/restricted_markdown_helper.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'redcarpet'
+require 'redcarpet/render_strip'
+
+# A markdown renderer that strips everything, except for codespans (via backticks)
+module RestrictedMarkdownHelper
```
Comment (Bjvanminnen) [inline dashboard/app/helpers/restricted_markdown_helper.rb]: Open to suggestions on whether there's a better name for this.

Comment (Hamms) [reply dashboard/app/helpers/restricted_markdown_helper.rb]: I like the specific name `CodeSpanOnly` for the renderer; why not maintain that throughout? `CodeSpanOnlyHelper` and `CodeSpanRenderer`?

File: dashboard/app/models/stage.rb [hunk 2]
```diff
@@ -106,8 +108,8 @@ def summarize
         flex_category: localized_category,
         lockable: !!lockable,
         levels: cached_script_levels.map(&:summarize),
-        description_student: (I18n.t "data.script.name.#{script.name}.stages.#{name}.description_student", default: ''),
-        description_teacher: (I18n.t "data.script.name.#{script.name}.stages.#{name}.description_teacher", default: '')
+        description_student: render_restricted_markdown(I18n.t("data.script.name.#{script.name}.stages.#{name}.description_student"), default: ''),
+        description_teacher: render_restricted_markdown(I18n.t("data.script.name.#{script.name}.stages.#{name}.description_teacher"), default: '')
```
Comment (Hamms) [inline dashboard/app/models/stage.rb]: isn't `default: ''` supposed to be an argument to I18n.t?

File: dashboard/app/helpers/restricted_markdown_helper.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/helpers/restricted_markdown_helper.rb]: Unless you want to keep the name generic to accommodate future spec changes, in which case I think as-is is fine.

File: dashboard/app/models/stage.rb (hunk 2 above)
Comment (Bjvanminnen) [reply dashboard/app/models/stage.rb]: nice catch. i originally didnt have parens, then the linter complained at me, so i added them (in the wrong spot) :man_facepalming:

**Pull request merged**